internal/graph/check.go

```diff
@@ -809,6 +809,12 @@ func (c *LocalChecker) checkUsersetFastPath(ctx context.Context, iter *storage.C
 	ctx, span := tracer.Start(ctx, "checkUsersetFastPath")
 	defer span.End()
 
+	typesys, ok := typesystem.TypesystemFromContext(ctx)
```

Comment:
IMO, by the time you reach here it's literally impossible for this to not be OK. I'd skip the check altogether.